apps/src/code-studio/components/progress/LessonProgress.jsx

```diff
@@ -160,16 +166,26 @@ class LessonProgress extends Component {
   }
 
   render() {
-    const {lessonExtrasUrl, onLessonExtras, lessonTrophyEnabled} = this.props;
-    let levels = this.props.levels;
+    const {
+      lessonExtrasUrl,
+      lessonTrophyEnabled,
+      levels,
+      studentProgress,
+      currentLevelId,
+      onLessonExtras
+    } = this.props;
 
+    // Bonus levels should not count towards mastery.
+    let filteredLevels = levels.filter(level => !level.bonus);
     // Only puzzle levels (non-concept levels) should count towards mastery.
     if (lessonTrophyEnabled) {
-      levels = levels.filter(level => !level.isConceptLevel);
+      filteredLevels = filteredLevels.filter(level => !level.isConceptLevel);
     }
 
-    // Bonus levels should not count towards mastery.
-    levels = levels.filter(level => !level.bonus);
+    const location = window.location.href;
+    const currentPage = parseInt(
+      location.substring(location.lastIndexOf('/') + 1)
```

Comment:
I think you'll also need to set an end point. urls can also have `?section=###` or `?enableExperiments=name`

On that note, this feels fragile. How do we set this number? Can we leverage that method to get this int?

We don't necessarily have to fix this to ship it, but a linked follow-up task would be nice.

Reply:
Here is a ticket to track this tech debt: https://codedotorg.atlassian.net/browse/LP-1641

From MDN: "If parseInt encounters a character that is not a numeral in the specified radix, it ignores it and all succeeding characters and returns the integer value parsed up to that point."

So luckily in this case this is not quite as fragile as it looks. Without a significant change to how we're doing things, the page number will always be the last path component of this url, and anything after it will be ignored. That said, I agree this is hacky, but it was the simplest solution I could think of.

Reply:
Oh cool! Thanks for the explanation - didn't realize that was how parseInt worked.